### PR TITLE
Remove timestamps from diagnostics output filenames

### LINKED ISSUE #1765: Better naming scheme for output?
### Description

Follow-up on the discussion with @jiahe23 @kpamnany @jakebolewski

Currently the output is named with an appended timestamp, which is inconvenient for scripted runs. What would be a better naming scheme?

### PR DESCRIPTION
**Pull request by kpamnany**
### Description

Now, any previously existing output files will be overwritten. Use `--output-dir` (or the environment variable or the keyword argument to `init`) to change the output directory if you don't want files overwritten.

Closes #1765.

- [X] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [X] Unit tests are included OR N/A.
- [X] Code is exercised in an integration test OR N/A.
- [X] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by kpamnany** [comment]
bors try

**Comment by ilopezgp** [comment]
LGTM

**Review by ilopezgp — APPROVED** [review]

**Comment by jakebolewski** [comment]
LGTM.  Somewhat related to this PR it might be use to have an `--overwite=true/false` flag as well which would through an error, that would be useful for deployments.

**Comment by jakebolewski** [comment]
Looks like one unit test expected the previous behavior: https://github.com/CliMA/ClimateMachine.jl/runs/1834568819#step:9:668

**Review by jakebolewski — APPROVED** [review]

File: src/InputOutput/Writers/netcdf_writer.jl [hunk 1]
```diff
@@ -18,8 +18,12 @@ function full_name(writer::NetCDFWriter, filename = nothing)
     end
 end
 
-function init_data(nc::NetCDFWriter, filename, dims, vars)
-    Dataset(full_name(nc, filename), "c") do ds
+function init_data(nc::NetCDFWriter, filename, no_overwrite, dims, vars)
+    nm = full_name(nc, filename)
+    if no_overwrite && ispath(nm)
+        error("$(nm) exists and overwriting is forbidden.")
```
Comment (jakebolewski) [inline src/InputOutput/Writers/netcdf_writer.jl:24]: maybe there should be a warning thrown when overwriting an existing file as well

**Review by jakebolewski — APPROVED** [review]

File: src/InputOutput/Writers/netcdf_writer.jl (hunk 1 above)
Comment (jakebolewski) [reply src/InputOutput/Writers/netcdf_writer.jl:24]: just following the principal of "least surprise" when debugging  :)

**Pull request merged**